auth/src/main/java/com/firebase/ui/auth/ui/email/RegisterEmailFragment.java

```diff
@@ -0,0 +1,253 @@
+ */
+public class RegisterEmailFragment extends BaseFragment implements
+        View.OnClickListener, View.OnFocusChangeListener {
+
+    public static final String TAG = "RegisterEmailFragment";
+
+    private EditText mEmailEditText;
+    private EditText mNameEditText;
+    private EditText mPasswordEditText;
+    private TextView mAgreementText;
+    private TextInputLayout mEmailInput;
+    private TextInputLayout mPasswordInput;
+
+    private EmailFieldValidator mEmailFieldValidator;
+    private PasswordFieldValidator mPasswordFieldValidator;
+    private RequiredFieldValidator mNameValidator;
+    private SaveSmartLock mSaveSmartLock;
+
+    public static RegisterEmailFragment getInstance(FlowParameters flowParameters,
+                                                    @Nullable String email,
+                                                    @Nullable String name) {
+        RegisterEmailFragment fragment = new RegisterEmailFragment();
+
+        Bundle args = new Bundle();
+        args.putParcelable(ExtraConstants.EXTRA_FLOW_PARAMS, flowParameters);
+        args.putString(ExtraConstants.EXTRA_EMAIL, email);
+        args.putString(ExtraConstants.EXTRA_NAME, name);
+
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater,
+                             @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+
+        View v = inflater.inflate(R.layout.register_email_layout, container, false);
+
+        mPasswordFieldValidator = new PasswordFieldValidator(
+                (TextInputLayout) v.findViewById(R.id.password_layout),
+                getResources().getInteger(R.integer.min_password_length));
+        mNameValidator = new RequiredFieldValidator(
+                (TextInputLayout) v.findViewById(R.id.name_layout));
+        mEmailFieldValidator = new EmailFieldValidator(
+                (TextInputLayout) v.findViewById(R.id.email_layout));
+
+        mEmailEditText = (EditText) v.findViewById(R.id.email);
+        mNameEditText = (EditText) v.findViewById(R.id.name);
+        mPasswordEditText = (EditText) v.findViewById(R.id.password);
+        mAgreementText = (TextView) v.findViewById(R.id.create_account_text);
+        mEmailInput = (TextInputLayout) v.findViewById(R.id.email_layout);
+        mPasswordInput = (TextInputLayout) v.findViewById(R.id.password_layout);
+
+        mEmailEditText.setOnFocusChangeListener(this);
+        mNameEditText.setOnFocusChangeListener(this);
+        mPasswordEditText.setOnFocusChangeListener(this);
+        v.findViewById(R.id.button_create).setOnClickListener(this);
+
+        // If email is passed in, fill in the field and move down to the name field.
+        String email = getArguments().getString(ExtraConstants.EXTRA_EMAIL);
+        if (!TextUtils.isEmpty(email)) {
+            mEmailEditText.setText(email);
```

Comment:
That's great! The user not being able to edit their email really bothered me so now that it's no longer the case... This change is going to be really cool especially with those animations!  👍